tests/yast2_gui/yast2_network_settings.pm

```diff
@@ -53,7 +53,13 @@ sub run {
     #	Hostname/DNS, set hostname via dhcp, yes for br0
     send_key 'alt-s';
     assert_screen 'yast2-network-settings_hostname-dns';
-    assert_and_click 'yast2-network-settings_hostname-dns_set-via-dhcp';
+    assert_screen([qw(yast2-network-settings_hostname-dns_set-via-dhcp dns_set-via-dhcp_no)], 90);
+    if (match_has_tag('dns_set-via-dhcp_no')) {
+        assert_and_click 'dns_set-via-dhcp_no';
```

Comment:
You linked  http://e13.suse.de/tests/4291#step/yast2_network_settings/ and the needle "dns_set-via-dhcp_no" there only contains the arrow to open the drop-down and will also match if this drop-down contains "any". Would be better if you needle the whole field (like "yast2-network-settings_hostname-dns_set-via-dhcp" has it) and the arrow in addition to that (assert_and_click always clicks on the last "box" you create in the needle)

Reply:
no, assert_and_click requires the exact area. there is no other way here. this works for a long time :)

Reply:
As I wrote above, assert_and_click uses the last box you created in your needle to click on it. This way you can make sure that the content of the needle is right and you can also make sure it clicks on the right area. I'll link a test after the meeting where I made use of this feature.